packages/cli/src/config/parseStoreConfig.ts

```diff
@@ -39,16 +84,26 @@ const TableDataFull = z
     return arg as RequireKeys<typeof arg, "dataStruct">;
   });
 
-const TableDataShorthand = FieldData.transform((fieldData) => {
-  return TableDataFull.parse({
+const ShorthandTableConfig = FieldData.transform((fieldData) => {
+  return FullTableConfig.parse({
     schema: {
       value: fieldData,
     },
   });
 });
 
-const TablesRecord = z.record(TableName, z.union([TableDataShorthand, TableDataFull])).transform((tables) => {
-  // default fileSelector depends on tableName
+export const TableConfig = FullTableConfig.or(ShorthandTableConfig);
+
+/************************************************************************
+ *
+ *    TABLES
+ *
+ ************************************************************************/
+
+export type TablesConfig = Record<string, TableConfig | z.input<typeof FieldData>>;
+
+export const TablesConfig = z.record(TableName, TableConfig).transform((tables) => {
+  // default route depends on tableName
```

Comment:
gonna rename `fileSelector` to `name` or sth soon, but for now let's use `fileSelector`
```suggestion
  // default fileSelector depends on tableName
```